browser/ui/views/commands/default_accelerators_mac.mm

```diff
@@ -0,0 +1,177 @@
+#include "base/logging.h"
+#include "base/strings/sys_string_conversions.h"
+#include "brave/app/command_utils.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/global_keyboard_shortcuts_mac.h"
+#include "chrome/browser/ui/cocoa/accelerators_cocoa.h"
+#include "ui/events/cocoa/cocoa_event_utils.h"
+#include "ui/events/event_constants.h"
+#include "ui/events/keycodes/keyboard_code_conversion_mac.h"
+
+static_assert(__OBJC__);
+static_assert(BUILDFLAG(IS_MAC));
+
+namespace commands {
+
+namespace {
+
+bool CanConvertToAcceleratorMapping(int command_id) {
+  return base::Contains(commands::GetCommands(), command_id);
+}
+
+bool CanConvertToAcceleratorMapping(NSMenuItem* item) {
+  if (!CanConvertToAcceleratorMapping(static_cast<int>(item.tag))) {
+    return false;
+  }
+
+  NSString* keyEquivalent = item.keyEquivalent;
+  return keyEquivalent != nil && [keyEquivalent length] > 0;
+}
+
+AcceleratorMapping ToAcceleratorMapping(NSMenuItem* item) {
+  NSString* keyEquivalent = item.keyEquivalent;
+  DVLOG(2) << __FUNCTION__ << item.tag << " > "
+           << base::SysNSStringToUTF16(keyEquivalent);
+  NSEvent* keyEvent = [NSEvent keyEventWithType:NSEventTypeKeyDown
+                                       location:NSZeroPoint
+                                  modifierFlags:0
+                                      timestamp:0
+                                   windowNumber:0
+                                        context:nil
+                                     characters:keyEquivalent
+                    charactersIgnoringModifiers:keyEquivalent
+                                      isARepeat:NO
+                                        keyCode:0];
+
+  auto modifiers = ui::EventFlagsFromModifiers(item.keyEquivalentModifierMask);
+  if (item.keyEquivalentModifierMask & NSEventModifierFlagFunction) {
+    // ui::EventFlagsFromModifiers doesn't handle the 'Function key.
+    modifiers |= ui::EF_FUNCTION_DOWN;
+  }
+
+  if (![keyEquivalent isEqualToString:[keyEquivalent lowercaseString]]) {
+    // Shift key could be omitted in |item.keyEquivalentModifierMask| even
+    // it's needed. In this case |item.keyEquivalent| can be already
+    // shifted/capitalized. So we should add it manually.
+    modifiers |= ui::EF_SHIFT_DOWN;
+  }
+
+  // "Close Tab" and "Close Window" item's shortcuts are statically the same.
+  // They are dynamically changed based on the context. e.g. has tab or not,
+  // is pwa or not. And also the value is hard coded.
+  // https://github.com/chromium/chromium/blob/299385e09d41d5ce3abd434879b5f9b0a8880cd7/chrome/browser/app_controller_mac.mm#L772
+  // Here we're to provide brave commands, manually add shift modifier for
+  // the "Close Window" item.
+  // TODO(sko) As the shortcut is hard coded in the file above, we might need to
+  // forbid to adjust default key for IDC_CLOSE_TAB and IDC_CLOSE_WINDOW.
+  if (static_cast<int>(item.tag) == IDC_CLOSE_WINDOW) {
+    modifiers |= ui::EF_SHIFT_DOWN;
```

Comment:
Interesting - the close window command always has shift down? Will this break if the user updates the shortcut for `Close Window` form the MacOS settings?